### PR TITLE
Add cookie storage driver.

### PR DESCRIPTION
**Pull request by dlongley**

### CODE REVIEW

File: cookieDriver.js [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+function _initStorage(options) {
+  const dbInfo = {};
+  if(options) {
+    Object.assign(dbInfo, options);
+  }
+  dbInfo.keyPrefix = _getKeyPrefix(options, this._defaultConfig);
+  // FIXME: support `serializer`?
+  //dbInfo.serializer = serializer;
+  this._dbInfo = dbInfo;
+  return Promise.resolve();
```
Comment (mattcollier) [inline cookieDriver.js]: If the intention here is to pave the way for a possible async operation in `_initStorage`, would making this an `async function` and then not doing this `Promise.resolve()` business be more clear?

File: cookieDriver.js [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+    await this.ready();
+    _clear(this._dbInfo.keyPrefix);
+  } catch(e) {
+    if(callback) {
+      return callback(e);
+    }
+    throw e;
+  }
+  if(callback) {
+    callback();
```
Comment (mattcollier) [inline cookieDriver.js:44]: add return for safety.

File: cookieDriver.js [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+      result = JSON.parse(result);
+    }
+  } catch(e) {
+    if(callback) {
+      return callback(e);
+    }
+    throw e;
+  }
+  if(callback) {
+    callback(null, result);
```
Comment (mattcollier) [inline cookieDriver.js:66]: add return for safety.  Addition instance of this pattern are not commented on.

File: cookieDriver.js [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+  if(callback) {
+    callback();
+  }
+}
+
+async function key(n, callback) {
+  let result = null;
+  try {
+    await this.ready();
+    const keyPrefix = this._dbInfo.keyPrefix;
```
Comment (mattcollier) [inline cookieDriver.js]: there are a number of places where destructured assignment could be used.

File: cookieDriver.js [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+  if(callback) {
+    callback(null, keys);
+  }
+  return keys;
+}
+
+async function length(callback) {
+  let length;
+  try {
+    const keys = this.keys();
```
Comment (mattcollier) [inline cookieDriver.js]: missing `await`

**Review by mattcollier — CHANGES_REQUESTED** [review]

File: cookieDriver.js (hunk 2 above)
Comment (dlongley) [reply cookieDriver.js:44]: The contract for this API (which is up to `localForage`) seems to be to return undefined, not a function.

File: cookieDriver.js (hunk 3 above)
Comment (dlongley) [reply cookieDriver.js:66]: See above -- and it's the same for the others. I copied what the other drivers from the `localForage` lib do.

**Comment by dlongley** [comment]
@mattcollier, comments addressed, ready for another pass. Note that the `cookieDriver` is copied from the other drivers in `localForage` and follows their pattern for callbacks, etc.

File: cookieDriver.js (hunk 2 above)
Comment (mattcollier) [reply cookieDriver.js:44]: does this mean that this would work?
```
callback();
return;
```

**Review by mattcollier — APPROVED** [review]

File: cookieDriver.js (hunk 2 above)
Comment (dlongley) [reply cookieDriver.js:44]: Yeah, but nothing happens after that ... so that leads me to think it would just be confusing/unnecessary.

**Pull request merged**